src/build-tools/common-grunt-rules.ts

```diff
@@ -215,4 +231,72 @@ export class Rule {
     return { files: allFilesForlibPaths.concat(copyInfo.files) };
   }
 
+  public buildAndRunTest(test :string, gruntConfig :{[c :string] :{[t :string] :Object}},
+                         coverage?:boolean) :string[] {
+    var name = test + 'Spec';
+    var browserifyRule = this.browserifySpec(test);
+    var jasmineRule = this.jasmineSingleSpec(test);
+
+    if (coverage) {
+      name += 'Cov';
+      browserifyRule = this.addCoverageToBrowserify(browserifyRule);
+      jasmineRule = this.addCoverageToSpec(jasmineRule);
+    }
+
+    gruntConfig['browserify'][name] = browserifyRule;
+    gruntConfig['jasmine'][name] = jasmineRule;
+
+    return [
+      'browserify:' + name,
+      'jasmine:' + name,
+    ];
+  }
+
+  /*
+   * Returns a list of tests that exist within the directory structure of a
+   * project.
+   *
+   * rootDir is the directory under which the layout will match what this file
+   * expects for paths being passed in (i.e. under devBuildPath)
+   *
+   * getTests('src');
+   *   Lists all tests under the src/ directory
+   * getTests('src', 'generic_ui/scripts');
+   *   Lists all tests under the generic_ui/scripts directory, all paths
+   *   relative to src/
+   * getTests('src', undefined, ['integration-tests']);
+   *   Lists all the tests under src/ ignoring anything named integration-tests
+   */
+  public getTests(rootDir :string, current?:string, ignore?:string[]) {
```

Comment:
As we discussed, let's leave that final parameter since you reckon it will be useful to many callers.